services/121-service/src/metrics/metrics.service.ts

```diff
@@ -426,11 +429,89 @@ export class MetricsService {
     const query = this.registrationScopedRepository
       .createQueryBuilder('registration')
       .select(`registration."registrationStatus" AS status`)
-      .addSelect(`COUNT(registration."registrationStatus") AS "statusCount"`)
+      .addSelect(`COUNT(registration."registrationStatus") AS "count"`)
       .andWhere({ programId })
       .andWhere({ registrationStatus: Not(RegistrationStatusEnum.deleted) })
       .groupBy(`registration."registrationStatus"`);
     const res = await query.getRawMany<RegistrationStatusStats>();
     return res;
   }
+
+  public async getRegistrationCountByDate(
+    programId: number,
+  ): Promise<Record<string, number>> {
+    const query = this.registrationScopedRepository
+      .createQueryBuilder('registration')
+      .select(`registration."created"::date`)
+      .addSelect(`COUNT(registration."referenceId")`)
+      .andWhere({ programId })
+      .groupBy(`"created"`)
+      .orderBy(`"created"`);
+    const res = (await query.getRawMany()).reduce(
+      (dates: Record<string, number>, r) => {
+        dates[new Date(r.created).toISOString()] = r.count;
+        return dates;
+      },
+      {},
+    );
+    return res;
+  }
+
+  public async getAllPaymentsAggregates(
+    programId: number,
+  ): Promise<Record<number, PaymentReturnDto>> {
+    const res: Record<number, PaymentReturnDto> = {};
+
+    const payments = (await this.paymentsService.getPayments(programId)).sort(
+      (pA, pB) => pA.payment - pB.payment,
+    );
+
+    for (const payment of payments) {
+      const aggregate = await this.paymentsService.getPaymentAggregation(
+        programId,
+        payment.payment,
+      );
+      res[payment.payment] = aggregate;
+    }
+
+    return res;
+  }
+
+  public async getAmountSentByMonth(
+    programId: number,
+  ): Promise<Record<string, Record<string, number>>> {
```

Comment:
this nested structure gets a bit too much I think. See earlier comment about making a dto.